Exercises/GerritFritz/computer.py

```
@@ -0,0 +1,98 @@
+        self.memory_pointer += 4
+
+    def inp(self):
+        try: self.commands[self.positions[0]] = self.UI.paddle_offset
+        except: raise RuntimeError("invalid input")
+        self.memory_pointer += 2
+
+    def out(self):
+        self.UI.screen_data.append(self.commands[self.positions[0]])
+        self.memory_pointer += 2
+
+    def jump_True(self):
+        if self.commands[self.positions[0]] != 0:  self.memory_pointer = self.commands[self.positions[1]]
+        else: self.memory_pointer += 3
+
+    def jump_False(self):
+        if self.commands[self.positions[0]] == 0: self.memory_pointer = self.commands[self.positions[1]]
+        else: self.memory_pointer += 3
+
+    def less(self):
+        if self.commands[self.positions[0]] < self.commands[self.positions[1]]: self.commands[self.positions[2]] = 1
+        else: self.commands[self.positions[2]] = 0
+        self.memory_pointer += 4
+        
+    def equals(self):
+        if self.commands[self.positions[0]] == self.commands[self.positions[1]]: 
+            self.commands[self.positions[2]] = 1
+        else: 
+            self.commands[self.positions[2]] = 0
+        self.memory_pointer += 4
+
+    def offset_increment(self):
+        self.offset += self.commands[self.positions[0]]
+        self.memory_pointer += 2
+
+    def mode_to_index(self):
+        self.positions  = []
+        for i in range(0,len(self.mode)):
+            if len(self.commands)>self.memory_pointer+i+1:
+                local_mode = self.mode[i]
+                match local_mode:
+                    case "0": self.positions.append(self.commands[self.memory_pointer+i+1])
+                    case "1": self.positions.append(self.memory_pointer+i+1)
+                    case "2": self.positions.append(self.offset+self.commands[self.memory_pointer+i+1])
+                    case _: raise KeyError(f"Wrong mode {self.mode}")
+            else:
+                raise IndexError(f"out of range")
```

Raising a index error with that description is kind of pointless since the code would just raise an IndexError on itself if you didn't sort out that case. You could at least provide more information here if you want to raise an error yourself. Examples of informaiton you could print here are the values of self.commands and self.memory_pointer.